sdk/core/azure-core/inc/azure/core/credentials.hpp

```diff
@@ -64,13 +64,13 @@ namespace Azure { namespace Core {
   /**
    * @brief An exception that gets thrown when authentication error occurs.
    */
-  class AuthenticationException : public std::runtime_error {
+  class AuthenticationException : public std::exception {
```

Comment:
Do we have tests that verify we are throwing AuthException from the right base type? We should add one since we want to prevent a future code change that changes this back to fail CI.

Reply:
This will be covered once #271 gets in.